h2o-core/src/main/java/water/server/ServletProvider.java

```diff
@@ -0,0 +1,24 @@
+package water.server;
+
+import javax.servlet.http.HttpServlet;
+import java.util.LinkedHashMap;
+
+public interface ServletProvider {
+
+  /**
+   * Provides a collection of Servlets that should be registered.
+   * @return a map of context path to a Servlet class
+   */
+  LinkedHashMap<String, Class<? extends HttpServlet>> servlets();
+
+  /**
+   * Provider priority, providers with higher priority will be used first. H2O Core Provider will be used last and
```

Comment:
What if two have the same priority ? What happens then ? From the impl, if it's not core, it'll get overriden in the order the services are discovered, which is not predictable behavior. Should we document it ?

Reply:
sure, more documentation is always better